nft_ingester/src/cleaners/fork_cleaner.rs

```diff
@@ -62,18 +62,19 @@ where
         Self { cl_items_manager, fork_checker, metrics }
     }
 
-    pub async fn clean_forks(&self, rx: Receiver<()>) {
+    pub async fn clean_forks(&self, cancellation_token: CancellationToken) {
         let last_slot_for_check =
             self.fork_checker.last_slot_for_check().saturating_sub(SLOT_CHECK_OFFSET);
-        let all_non_forked_slots = self.fork_checker.get_all_non_forked_slots(rx.resubscribe());
+        let all_non_forked_slots =
+            self.fork_checker.get_all_non_forked_slots(cancellation_token.child_token());
 
         let mut forked_slots = 0;
         let mut delete_items = Vec::new();
 
         // from this column data will be dropped by slot
         // if we have any update from forked slot we have to delete it
         for cl_item in self.cl_items_manager.cl_items_iter() {
-            if !rx.is_empty() {
+            if cancellation_token.is_cancelled() {
```

Comment:
The huge function is prevented from shutting down by that cancellation because it is happening only at the function beginning.

Reply:
will be reviewed in the next iterations of this PR